openapi.yaml

```diff
@@ -373,62 +465,122 @@ components:
           type: string
-          description: ID of the specific message being rated
-        feedback:
-          $ref: '#/components/schemas/Feedback'
-        suggestedChanges:
+          description: External ID of the thread containing the target message
+        externalMessageId:
           type: string
-          description: Optional proposed changes to the message content
+          description: ID of the specific message for feedback. If not provided, feedback will be attached to the thread's last message
           nullable: true
-        user:
-          $ref: '#/components/schemas/UserInfo'
+        projectId:
+          type: integer
+          description: Recommended - ID of the project this thread belongs to. Helps ensure correct thread identification
+          nullable: true
+        externalUser:
+          $ref: '#/components/schemas/CreateExternalUserRequest'
+          nullable: true
+          description: External user information for associating feedback with a user
+        attributes:
+          type: object
+          description: |
+            Key-value pairs where:
+            - key: attribute name (must match an existing attribute in the project)
+            - value: option name (must match an existing option for that attribute)
+          additionalProperties:
+            type: string
+          example:
+            Quality: "Wonderful"
+            Relevance: "High"
+      required:
+        - feedbackType
+        - externalThreadId
+
+    ExternalFeedbackMessageAssociation:
+      type: object
+      properties:
+        id:
+          type: integer
+          format: int32
+          description: Unique identifier for the message association
+        messageId:
+          type: integer
+          format: int32
+          description: ID of the associated message
+        externalFeedbackId:
+          type: integer
+          format: int32
+          description: ID of the external feedback (unique)
+        createdAt:
+          type: string
+          format: date-time
+        updatedAt:
+          type: string
+          format: date-time
       required:
-        - threadId
+        - id
         - messageId
-        - feedback
+        - externalFeedbackId
 
     ExternalFeedback:
       type: object
       properties:
         id:
-          type: string
+          type: integer
+          format: int32
+          description: Unique identifier for the feedback
         feedbackType:
           type: string
+          maxLength: 255
+          nullable: true
+          description: Type of the feedback
         feedbackText:
           type: string
-        threadId:
-          type: string
-        messageId:
-          type: string
+          nullable: true
+          description: Text content of the feedback
         userId:
+          type: integer
+          format: int32
+          nullable: true
+          description: ID of the internal user if applicable
+        externalUserId:
```

Comment:
I've also updated this to now include the ExternalUser object in the return object but not the id. An example response now looks like this:

```
{
  "id": 313824,
  "feedbackType": "POSITIVE",
  "feedbackText": "This is great",
  "createdAt": "2024-11-08T22:38:01.147Z",
  "updatedAt": "2024-11-08T22:38:01.147Z",
  "externalUser": {
    "id": 4708,
    "organizationId": 1,
    "externalId": "mintlify-test-1",
    "name": "Mr Bob",
    "email": null,
    "username": null,
    "createdAt": "2024-11-08T19:55:35.728Z",
    "updatedAt": "2024-11-08T22:23:50.350Z",
    "segments": []
  },
  "attributeOptions": [
    {
      "id": 39,
      "attributeId": 22,
      "name": "Low",
      "createdAt": "2024-11-08T22:38:00.991Z",
      "updatedAt": "2024-11-08T22:38:00.991Z",
      "attribute": {
        "id": 22,
        "organizationId": 1,
        "projectId": 12074,
        "name": "Humor Level",
        "createdAt": "2024-11-08T22:38:00.952Z",
        "updatedAt": "2024-11-08T22:38:00.952Z"
      }
    }
  ]
}
```